pyccel/parser/semantic.py

```diff
@@ -3274,6 +3293,7 @@ def _visit_FunctionDef(self, expr):
                 arguments = arguments[1:]
                 dt        = self.get_class_construct(cls_name)()
                 cls_base  = self.scope.find(cls_name, 'classes')
+                cls_base.scope.insert_symbols(expr.scope.local_used_symbols.copy())
```

Comment:
Shouldn't this be in `_visit_ClassDef`?

Reply:
It's the same things if it's in the `_visit_ClassDef` and I think it's better to let it in the `_visit_FunctionDef`.

Reply:
Why do you think it is better here? I think in `_visit_ClassDef` it can be done at the creation of the scope. It will only need to be done once and an `if` check is not needed

Reply:
Ah! Wait I think I see. Here you are not updating with the class scope variables but rather with the local variables of the function.
Can you explain why this is needed please

Reply:
I need it for : https://github.com/pyccel/pyccel/blob/e7b19d20720e2c2800a7858162928c0aae6ea639/pyccel/parser/semantic.py#L1156-L1156

Reply:
Hmm, these shouldn't be the same thing but I'm not sure how to fix it. Let's leave it for now

Reply:
The `get_expected_name()` check's if the `attribute_name` is in the `class_def.scope._used_symbols` That's why I insert them into the `classdef.scope`
https://github.com/pyccel/pyccel/blob/2fb0e0da37382e0ed8c144865547a70bcbb14b8c/pyccel/parser/scope.py#L560-L567

Reply:
```
class Point:
     def __init__(self):
         self.x = 1
         self.X = 1

    def get_x(self):
        return self.x

    def get_X(self):
        return self.X
```

Reply:
Can you test this example and open an issue containing the generated code if it fails please. I suspect it will generate code that won't compile. I don't have a computer to run it myself

Reply:
It returns an Error, I'll open an issue with what should be fixed.